Libplanet/Blockchain/BlockChain.cs

```diff
@@ -265,18 +265,42 @@ bool IsPossiblyIn(BitArray addr, BitArray maskPattern)
         public void Append(Block<T> block, DateTimeOffset currentTime) =>
             Append(block, currentTime, render: true);
 
-        public void StageTransactions(ISet<Transaction<T>> txs)
+        /// <summary>
+        /// Adds <paramref name="transactions"/> to the pending list so that
+        /// a next <see cref="Block{T}"/> to be mined contains these
+        /// <paramref name="transactions"/>.
+        /// </summary>
+        /// <param name="transactions">
+        /// <see cref="Transaction{T}"/>s to add to the pending list.</param>
+        public void StageTransactions(ISet<Transaction<T>> transactions)
         {
-            foreach (Transaction<T> tx in txs)
+            foreach (Transaction<T> tx in transactions)
             {
                 Transactions[tx.Id] = tx;
             }
 
             Store.StageTransactionIds(
-                txs.Select(tx => tx.Id).ToImmutableHashSet()
+                transactions.Select(tx => tx.Id).ToImmutableHashSet()
             );
         }
 
+        /// <summary>
+        /// Removes <paramref name="transactions"/> from the pending list.
+        /// </summary>
+        /// <param name="transactions"><see cref="Transaction{T}"/>s
+        /// to remove from the pending list.</param>
+        /// <seealso cref="StageTransactions"/>
+        public void UnstageTransactions(ISet<Transaction<T>> transactions)
+        {
+            foreach (Transaction<T> tx in transactions)
+            {
+                Transactions.Remove(tx.Id);
+            }
```

Comment:
@longfin Should we remove these transactions from pool as well?

This could break things, for example:

```csharp
b = Block<Action>.Mine(..., transactions: txs);
blockChain.Append(b);
blockChain.UnstageTransactions(txs);
Block b2 = blockChain[b.blockHash];  // This will crash.
```

Reply:
Yes, it can be problem. and, I think that `UnstageTransactions()` should remove only staged transactions, not mined transaction.

Reply:
You're right. So should we make like `BlockChain<T>.UnstageTransactionIds()` instead of `BlockChain<T>.UnstageTransactions()`?

Reply:
The current name looks better to me, because it exposes less implementation details.